routes/users.js

```diff
@@ -1,9 +1,100 @@
-const express = require('express');
-const router = express.Router();
+const express = require('express')
+const { User, db } = require('../database/User.js')
+const router = express.Router()
+const mongoose = require('mongoose')
 
-/* GET users listing. */
-router.get('/', (request, response, next) => {
-  response.send('respond with a resource');
-});
 
-module.exports = router;
+router.get('/', ( request, response, next ) => {
+  response.send( 'respond with a resource' )
+})
+
+router.post('/', ( request, response, next ) => {
+  const { username } = request.body
+  const newUser = new User({
+    username: username,
+    dashboards: {
+       widgets: {}
+    }
+  })
+  newUser.save( ( error ) => {
+    if( error ) return handleError( error )
+  })
+})
+
+router.get('/widgets/:widgetId', ( request, response, next ) => {
+  const { widgetId } = request.params
+
+  User.find( {'widgets._id': widgetId},function(error, widgets){
+      return widgets
+  })
+  .then( data => console.log('finished') )
+})
+
+router.post('/:userId/widgets', (request, response, next ) => {
+  const { userId } = request.params
+  const widget = { type, title, size, contents }
+  widget = request.body
+
+  const widgetAttributes = {
+    type: type,
+    title: title,
+    size: size,
+    contents: contents
+  }
+
+  User.findOneAndUpdate(
+    { _id: userId },
+    {
+      $push: {
+        widgets: widgetAttributes
+      }
+    },
+    ( error ) => {
+      if( error ) console.log( error )
+    }
+  )
+})
+
+router.put('/:userId/widgets/:widgetId', ( request, response, next ) => {
```

Comment:
Omit route; it does not make sense in the context of our domain (users do not contain widgets).